doc/edgehub-api.md

```diff
@@ -0,0 +1,434 @@
+- The token generation can be implemented by [program code](https://github.com/microsoft/azure-docs/blob/master/articles/iot-hub/iot-hub-dev-guide-sas.md)
+
+If there is an MQTT client library where an MQTT client object has a Connect() method with the signature:
+```
+mqttClient.Connect(string clientId, string username, string password)
+```
+
+Then using this method to connect a device named 'TestDevice' using an IoT Hub named 'test-srv', a possible call is:
+
+```
+mqttClient.Connect(
+                "TestDevice",
+                "test-srv.azure-devices.net/TestDevice?api-version=2018-06-30",
+                "SharedAccessSignature sr=test-srv.azure-devices.net%2Fdevices%2FTestDevice&sig=Mzv6sxyYH6WWfCpnYtkQBaF0IwfTPJFOULfWB80BEvQ%3D&se=1637291060")
+```
+Edge Hub supports the Clean Session flag in the MQTT packet. If it is not set and a device or module had a session previously, then Edge Hub restores the previous subscriptions (e.g. for receiving twin patches)
+### Sending Telemetry
+Sending telemetry message is the most common and simplest operation. Edge Hub supports sending using Qos0 and Qos1. In order to send telemetry:
+- devices can use the topic: devices/{device_id}/messages/events
+- modules can use the topic: devices/{device_id}/modules/{module_id}/messages/events, where device_id is the name of the Edge device
+
+Telemetry messages can encode properties similarly than URL query strings. This section is called 'property bag' and must be encoded according to [RFC2396](https://www.ietf.org/rfc/rfc2396.txt). The following example adds two parameters to the telemetry message; the first parameter name is called "food" and its value is "I like fries" and the second parameter name is "pet" and its value is "I like cats":
+```
+devices/test_device/messages/events/?food=I%20like%20fries&pet=I%20like%20cats
+```
+The properties can be retrieved as user properties by the back end application. Note, that Edge Hub/IoT Hub may add additional system properties to the message.
+
+When Edge Hub receives a telemetry message and it is QoS1, it acknowledges the message immediately. It does not mean that the message will be forwarded right away. Edge Hub stores the message, it runs the message through its [routing](https://docs.microsoft.com/en-us/azure/iot-edge/module-composition?view=iotedge-2020-11#declare-routes) and then it decides what to do. If Edge Hub is currently connected to IoT Hub and the message is routed upstream, then the message is going to be forwarded upstream. Otherwise the message may be sitting in the internal store of Edge Hub until the target becomes available. It is also possible that the routing does not device any target for the message and it gets dropped.
```

Comment:
```suggestion
When Edge Hub receives a telemetry message and it is QoS1, it acknowledges the message immediately. It does not mean that the message will be forwarded right away. Edge Hub stores the message, it runs the message through its [routing](https://docs.microsoft.com/en-us/azure/iot-edge/module-composition?view=iotedge-2020-11#declare-routes) and then it decides what to do. If Edge Hub is currently connected to IoT Hub and the message is routed upstream, then the message is going to be forwarded upstream. Otherwise the message may be sitting in the internal store of Edge Hub until the target becomes available. It is also possible that the routing does not target any device for the message and it gets dropped.
```